core/src/main/java/io/grpc/internal/AbstractStream.java

```diff
@@ -188,11 +189,15 @@ protected final void closeDeframer(boolean stopDelivery) {
      * Called to parse a received frame and attempt delivery of any completed messages. Must be
      * called from the transport thread.
      */
-    protected final void deframe(final ReadableBuffer frame) {
+    protected final void deframe(ReadableBuffer frame) {
+      Status s;
       try {
-        deframer.deframe(frame);
+        s = deframer.deframe(frame);
       } catch (Throwable t) {
```

Comment:
If we could end up removing catches like this, things may become more clear.

Reply:
Unfortunately that would be a behavior change.